### ISSUE TITLE
Source maps not found.  Can debug Renderer but not Main

### ISSUE THREAD

**Issue by greg9504** [Microsoft/vscode-node-debug#152]
VS Code Version:

> Version 1.13.1
> Commit 379d2efb5539b09112c793d3d9a413017d736f89
> Date 2017-06-14T18:21:47.485Z
> Shell 1.6.6
> Renderer 56.0.2924.87
> Node 7.4.0

When I attempt to debug a main process of an Electron app that was built with webpack it can't seem to find the source files, even though the trace log shows that the mapping was found. I am unable to set breakpoints. This is the error shown in the log:

```
20820: setBreakPointsRequest: {"path":"d:\\devtools\\js\\tutorials\\aeds_clean\\src\\electron\\index.ts","name":"index.ts"} [{"line":33}]
20820: _updateBreakpoints: result [{"verified":false,"message":"Breakpoint ignored because generated code not found (source map problem?)."}]
```

However earlier in the log I see this:
```
20820: _findSourceMapUrl: source map url found at end of generated file 'd:/devtools/js/tutorials/aeds_clean/dist/index.js'
...
20820: _registerSourceMap: webpack:///src/electron/index.ts -> d:/devtools/js/tutorials/aeds_clean/dist/index.js
```
which seems to suggest that the correct mapping has been found.

This is my launch configuration, Debug Main Process:
```
  {
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Debug Main Process",
            "type": "node",
            "request": "launch",
            "cwd": "${workspaceRoot}/dist",
            "trace": true,
            "sourceMaps": true,
            //"runtimeExecutable": "${workspaceRoot}/node_modules/.bin/electron",
            // Use the following for Windows
            "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/electron.cmd",
            "outFiles": [
                "${workspaceRoot}/dist/**/*.js"
            ],
            "program": "${workspaceRoot}/dist/index.js",
            "protocol": "legacy"
        },
        {
            "name": "Debug Renderer Process",
            "type": "chrome",
            "request": "launch",
            "trace": true,
            "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/electron",
            "windows": {
                "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/electron.cmd"
            },
            "runtimeArgs": [
                "${workspaceRoot}/dist",
                "--enable-logging",
                "--remote-debugging-port=9222"
            ],
            "sourceMaps": true,
            "webRoot": "${workspaceRoot}"
        }
    ]
}
```
I swear this used to work.  Not sure what version this stopped working.

What is a bit confusing is that I can debug the Renderer process no problem using the Debugger for Chrome extension, breakpoints set in .ts files are hit. I've attached the trace logs for both situations and the .script output for the renderer.
Thanks
[sourcemapsproblem2.zip](https://github.com/Microsoft/vscode-node-debug/files/1113251/sourcemapsproblem2.zip)

**Comment by greg9504** [comment]
OK think this is a "not supported" issue.  After reading:
https://code.visualstudio.com/docs/nodejs/nodejs-debugging#_javascript-source-map-tips
I see that the built in Node debugger for VS Code does not support the webpack:/// prefix, which my source maps had.  I added the devtoolModuleFilenameTemplate and devtoolFallbackModuleFilenameTemplate settings to the webpack "output" config so that the webpack:/// wasn't added to the path.  In my case I had to tweak it a bit and also add "..//" as my project was built to a "dist" directory.  Here's the complete output config:

```
output: {
      path: DEV_SERVER ? helpers.root('dev') : helpers.root('dist'),
      filename: '[name].js',
      sourceMapFilename: '[file].map',
      devtoolModuleFilenameTemplate: "..//[absolute-resource-path]",
      devtoolFallbackModuleFilenameTemplate: "..//[absolute-resource-path]?[hash]"      
    },
```
